crates/pixi_command_dispatcher/src/command_dispatcher/mod.rs

```diff
@@ -625,19 +625,50 @@ impl CommandDispatcher {
     /// - For URL sources: Extracts the archive with the exact checksum
     ///   (unimplemented)
     pub async fn checkout_pinned_source(
+        &self,
+        pinned_spec: &PinnedSourceSpec,
+    ) -> Result<SourceCheckout, CommandDispatcherError<SourceCheckoutError>> {
+        self.checkout_pinned_spec(pinned_spec.clone(), None).await
+    }
+
+    /// Checkout a source described by a [`SourceCodeLocation`], automatically
+    /// falling back to the manifest path when no dedicated build source is set.
+    ///
+    /// This is useful for the case where you want to checkout a source that is at
+    /// different location than the manifest, most obviously in an out-of-tree build.
+    pub async fn checkout_source_location(
+        &self,
+        source: &SourceCodeLocation,
+    ) -> Result<SourceCheckout, CommandDispatcherError<SourceCheckoutError>> {
+        let (primary, alternative_root) = source.as_source_and_alternative_root();
+        self.checkout_pinned_spec(primary.clone(), alternative_root.cloned())
+            .await
+    }
+
+    async fn checkout_pinned_spec(
         &self,
         pinned_spec: PinnedSourceSpec,
+        alternative_root: Option<PinnedSourceSpec>,
     ) -> Result<SourceCheckout, CommandDispatcherError<SourceCheckoutError>> {
         match pinned_spec {
-            PinnedSourceSpec::Path(ref path) => {
+            PinnedSourceSpec::Path(path_spec) => {
+                let alternative_root_path = match alternative_root.as_ref() {
```

Comment:
I think this can be replaced with `SourceAnchor`! Its purpose is to make one sourcespec relative to another.

You can also have a git source with a subdirectory for instance. I dont think the current implementation deals with that?